### PR TITLE
Adapt defact-test-output style to complicate test

### LINKED ISSUE #126: onnx_chainer.export_testcase does not like cupy.arrays
```
import numpy as np
import chainer
import chainer.functions as F

import onnx_chainer


class Model(chainer.Chain):
    def forward(self, x):
        return F.relu(x)


model = Model()
model.to_gpu()
x = model.xp.array([2,3], dtype=np.float32)
onnx_chainer.export_testcase(model, [x], 'o')
```

An exception will be thrown:

```
$ python3 bug2.py
Traceback (most recent call last):
  File "bug2.py", line 16, in <module>
    onnx_chainer.export_testcase(model, [x], 'o')
  File "/home/i/wrk/onnx-chainer/onnx_chainer/export_testcase.py", line 49, in export_testcase
    t = numpy_helper.from_array(var.data, 'Input_%d' % i)
  File "/home/i/.local/lib/python3.6/site-packages/onnx/numpy_helper.py", line 104, in from_array
    tensor.raw_data = arr.tobytes()  # note: tobytes() is only after 1.9.
AttributeError: 'cupy.core.core.ndarray' object has no attribute 'tobytes'
```

### PR DESCRIPTION
**Pull request by disktnk**
- `test_rt.py` and `test_input.py` are applied `ONNXModelTest`, these tests also make "model.onnx" + test_data_sets, and check output value by test runtime
- `test_export_testcase.py` is fixed not to put ONNX files to "out" directory, use temporary directory instead. Output files are not used for other test tool.
- fixed onnxruntime and onnx version
- fixed #126

### CODE REVIEW

File: onnx_chainer/export_testcase.py [hunk 1]
```diff
@@ -41,20 +40,22 @@ def export_testcase(
...
-            t = numpy_helper.from_array(var.data, 'Input_%d' % i)
+            array = chainer.cuda.to_cpu(var.array)
+            t = numpy_helper.from_array(array, 'Input_%d' % i)
             f.write(t.SerializeToString())
 
     for i, var in enumerate(outputs):
         with open(os.path.join(test_data_dir, 'output_%d.pb' % i), 'wb') as f:
-            t = numpy_helper.from_array(var.data, '')
+            array = chainer.cuda.to_cpu(var.array)
+            t = numpy_helper.from_array(array, '')
```
Comment (shinh) [inline onnx_chainer/export_testcase.py]: Maybe better to set actual output names. TODO would be fine (I think we would need to refactor code around here when we support custom input/output names).

File: onnx_chainer/export_testcase.py [hunk 2]
```diff
@@ -41,20 +40,22 @@ def export_testcase(
     os.makedirs(test_data_dir, exist_ok=True)
     for i, var in enumerate(inputs):
         with open(os.path.join(test_data_dir, 'input_%d.pb' % i), 'wb') as f:
-            t = numpy_helper.from_array(var.data, 'Input_%d' % i)
+            array = chainer.cuda.to_cpu(var.array)
+            t = numpy_helper.from_array(array, 'Input_%d' % i)
```
Comment (shinh) [inline onnx_chainer/export_testcase.py]: Maybe better to use `_get_graph_input_names`, but TODO would be fine.

File: onnx_chainer/export_testcase.py [hunk 3]
```diff
@@ -41,20 +40,22 @@ def export_testcase(
...
     if output_grad:
         # Perform backward computation
         if len(outputs) > 1:
             outputs = chainer.functions.identity(*outputs)
         for out in outputs:
-            out.grad = np.ones_like(out.data)
+            out.grad = model.xp.ones_like(out.data)
         outputs[0].backward()
 
         for i, (name, param) in enumerate(model.namedparams()):
```
[the remark this answers is not in the document]
Comment (disktnk) [reply onnx_chainer/export_testcase.py]: - add `write_tensor_pb` on "onnx_helper.py", and use it for these writing.
- adapt `chainer.cuda.to_cpu` on grad

File: onnx_chainer/export_testcase.py (hunk 1 above)
Comment (disktnk) [reply onnx_chainer/export_testcase.py]: add TODO

File: onnx_chainer/export_testcase.py (hunk 2 above)
Comment (disktnk) [reply onnx_chainer/export_testcase.py]: copied `_get_graph_input_names`, add TODO, refactor them on adding `input_names` and `output_names` support.

**Pull request merged**